query-compiler/query-compiler/tests/snapshots/queries__queries@create-nested-create-with-composite-id.json.snap

```diff
@@ -0,0 +1,46 @@
+---
+source: query-compiler/query-compiler/tests/queries.rs
+expression: pretty
+input_file: query-compiler/query-compiler/tests/data/create-nested-create-with-composite-id.json
+---
+let 0 = unique (query «INSERT INTO "public"."ParentModelWithCompositeId"
+                       ("a","b") VALUES ($1,$2) RETURNING
+                       "public"."ParentModelWithCompositeId"."a",
+                       "public"."ParentModelWithCompositeId"."b"»
+                params [const(BigInt(1)), const(BigInt(1))])
+in let 0$a = mapField a (get 0);
+       0$b = mapField b (get 0)
+   in sum (execute «INSERT INTO "public"."ChildOfModelWithCompositeId"
+                    ("id","parentA","parentB") VALUES ($1,$2,$3), ($4,$5,$6)»
+           params [const(BigInt(1)), var(0$a as Int), var(0$b as Int),
+                   const(BigInt(2)), var(0$a as Int), var(0$b as Int)]);
+   let 2 = let 0$a = mapField a (get 0);
+               0$b = mapField b (get 0)
+       in let @parent = unique (query «SELECT
+                                       "public"."ParentModelWithCompositeId"."a",
+                                       "public"."ParentModelWithCompositeId"."b"
+                                       FROM
+                                       "public"."ParentModelWithCompositeId"
+                                       WHERE
+                                       ("public"."ParentModelWithCompositeId"."a"
+                                       = $1 AND
+                                       "public"."ParentModelWithCompositeId"."b"
+                                       = $2) LIMIT $3 OFFSET $4»
+                                params [var(0$a as Int), var(0$b as Int),
+                                        const(BigInt(1)), const(BigInt(0))])
+          in let @parent$b = mapField b (get @parent);
+                 @parent$a = mapField a (get @parent)
```

Comment:
Unrelated, thinking about optimizations again: we should probably mark each query node with the original model the query was generated for. Then, during one of the optimization passes, we could check that if we already have bindings to the same fields from this model in scope and we found no write query to this model while walking the tree up to those bindings, the query is redundant and we can eliminate it.